Töchtaüse Juicer Fix

Shadyyy66 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Töchtaüse berries haven't been juicable for a while, so I fixed that.

Removed grind_results since it's redundant and removed the itching powder reagent from the juice_results variable. Added nutriment to Töchtaüse berries for juicing purposes.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->
When you JUICE something it should be JUICED! got damn!!!

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: Töchtaüse berries are now juiceable. You no longer get Töchtaüse juice via grinding them.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Shadyyy66 committed:
Töchtaüse berry juicer fix

removed grind_results since it's redundant and removed the itching powder reagent from the juice_results variable. Added nutriment to Töchtaüse berries for juicing purposes

Mothblocks commented:
Can you explain the fix to me? Is having both grind_results and juice_results causing issues?

Shadyyy66 commented:
> Can you explain the fix to me? Is having both grind_results and juice_results causing issues?

the issue isnt that it has both grind_results and juice_results, the issue is that the way juicing works in SS13 is that it converts nutriment in the plant to whatever juice_results is set to. The problem was that Töchtaüse berries didn't have any nutriment in it, so I went ahead and gave it some.

I also removed the grind_results since it's redundant and just gives you the reagents that are already inside the plant (itching powder, the nutriment i added, etc)

Mothblocks commented:
So the bug was we had juice_results set but no nutriment? Is that something that is ever correct? I ask because I wonder if we (not necessarily you) can make this an automated test

Shadyyy66 commented:
i dont understand the question, but yes, the plant had no nutriment so juice_results could never create juice, which was clearly the intent since they had to go out of their way to add it to grind_results

Mothblocks commented:
If my question doesn't make sense then I probably don't understand juicing.

My question is, is there ever a time where juice_results is set, but nothing has nutriment, AND it's not a bug?

Shadyyy66 commented:
> My question is, is there ever a time where juice_results is set, but nothing has nutriment, AND it's not a bug?

juice_results requires nutriment conversion to function, if a plant without nutriment is juiced the results will be nothing, so to my knowledge, no, I don't think anything is made with the intent to purposely give you nothing when juiced when juice_results is set, otherwise juice_results would simply not be set in the first place.

Mothblocks commented:
Cool
